### DISCUSSION TITLE
TCXO Control DIO3 causing RXTimeOut IRQ in SX 1262

### DISCUSSION THREAD

**Discussion by prabwyld** [Lora-net/LoRaMac-node#1049] [Q&A]
Hello,

I have been using the DIO3 control for TCXO. The delay is being set to 31.25us. The radio is configured to work as a receiver and set to continuous mode. However the radio is firing RxtimeOut IRQ. I have checked the configuration and all IRQ's are being assigned to DIO1. Any idea what can be triggering an rxtimeOut.

#define TCXO_TIMEOUT_MS      0x02
void SX126xIoTcxoInit( void )
{
	SX126xSetDio3AsTcxoCtrl(TCXO_CTRL_3_3V, TCXO_TIMEOUT_MS);
}

P.s : If I ignore the first timeout and set it receiver again subsequently then the IRQ is not fired and it works fine.

**Comment by prabwyld** [comment]
Hello,

Could anyone please give some suggestions as to what may be causing the problem of spurious RX timeout IRQ.

**Reply by mluis1** [reply]
A TCXO requires a lot more time to wake up than 31.25us.

On `sx1262mbxdas-board.c` example implementation you can see below code

https://github.com/Lora-net/LoRaMac-node/blob/9a6c7dc02e01282c0d6d9102196abe7925e81a49/src/boards/NucleoL476/sx1262mbxdas-board.c#L97-L104

The `BOARD_TCXO_WAKEUP_TIME` definition is set to 5 ms. This value is suitable in most situations. In case a more precise value is required I would recommend to look at your TCXO datasheet in order to know the precise wake up time.
